js/modules/k6/experimental/tracing/client.go

```diff
@@ -93,9 +99,7 @@ func (c *Client) Configure(opts options) error {
 	return nil
 }
 
-// Request instruments the http module's request function with tracing headers,
-// and ensures the trace_id is emitted as part of the output's data points metadata.
-func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error) {
+func (c *Client) addTraceIDToArgs(args []goja.Value) ([]goja.Value, error) {
```

Comment:
Good call on attempting to extract the logic!

However, I think the function now does too much, also some things unrelated to modifying the arguments in place (we already have the `instrumentArguments` function to do that specifically. 

In the spirit of factorizing and making this logic reusable across the two request functions, I'd prefer it if instead of this specific function, we would move the trace id generation logic to a dedicated method: `Client.generateTraceContext`, and keep the setting of metadata external to that. Something along the lines of:
```javascript
func (c *Client) generateTraceContext() (header http.Header, traceID string, err error) {
	traceID = TraceID{
		Prefix: k6Prefix,
		Code:   k6CloudCode,
		Time:   time.Now(),
	}

	encodedTraceID, err = traceID.Encode()
	if err != nil {
		return http.Header{}, TraceID{}, fmt.Errorf("failed to encode the generated trace ID; reason: %w", err)
	}

	// Produce a trace header in the format defined by the
	// configured propagator.
	traceContextHeader, err = c.propagator.Propagate(traceID)
	if err != nil {
		return http.Header{}, TraceID{}, fmt.Errorf("failed to propagate trace ID; reason: %w", err)
	}
	
	return traceContextHeader, traceID, nil
}

func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*httpmodule.Response, error) {
    traceHeader, traceID, err = c.generateTraceContext()
    if err != nil {
        return nil, err
    }

    args, err = c.instrumentArguments(traceHeader, args)
	if err != nil {
		return nil, err
	}
	
	// Add the trace ID to the VU's state, so that it can be
	// used in the metrics emitted by the HTTP module.
	c.vu.State().Tags.Modify(func(t *metrics.TagsAndMeta) {
		t.SetMetadata(metadataTraceIDKeyName, encodedTraceID)
	})
	
	// Remove the trace ID from the VU's state, so that it doesn't
	// leak into other requests.
	defer func() {
		c.vu.State().Tags.Modify(func(t *metrics.TagsAndMeta) {
			t.DeleteMetadata(metadataTraceIDKeyName)
		})
	}()
	
	return c.requestFunc(method, url, args...) 
}

// same structure for asyncRequest
```

🙇🏻

Reply:
I'm happy with that 😄 🚀 🙇🏻